agent/consul/filter.go

```diff
@@ -25,7 +25,7 @@ func (d *dirEntFilter) Move(dst, src, span int) {
 func FilterDirEnt(acl acl.ACL, ent structs.DirEntries) (structs.DirEntries, error) {
 	df := dirEntFilter{acl: acl, ent: ent}
 	filtered := ent[:FilterEntries(&df)]
-	if len(filtered) == 0 {
```

Comment:
this by itself was causing valid acls but missing keys to return a 403, should have caught that in unit tests.

I added test cases for this now in filter_test.go